apps/s_server.c

```diff
@@ -3330,6 +3348,43 @@ static int www_body(int s, int stype, int prot, unsigned char *context)
                              "HTTP/1.0 200 ok\r\nContent-type: text/plain\r\n\r\n");
             }
             /* send the file */
+#ifndef OPENSSL_NO_KTLS
+            if (use_sendfile) {
+                FILE *fp = NULL;
+                int fd;
+                struct stat st;
+                off_t offset = 0;
+                size_t sz;
+
+                BIO_get_fp(file, &fp);
+                fd = fileno(fp);
+                if (fstat(fd, &st) < 0) {
+                    BIO_printf(io, "Error stat '%s'\r\n", p);
```

Comment:
It seems this kind of errors should not be printed into BIO `io`

Reply:
From my understandings, the informative messages need to be sent back to the client. I am not sure if internal errors like this and line 3373 are suitable to send out on the network...

Reply:
I took a closer look again. The error message here is to be output to BIO io, where io is the client. The BIO_new_file () call above this code fails and also outputs an error message to the client.